.github/workflows/ci.yml

```diff
@@ -0,0 +1,33 @@
+name: Java CI with Maven
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  workflow_dispatch:
```

Comment:
What about simply having:
```yml
on: [push, pull_request]
```
which would trigger the build for any branch?

Also, not sure you really need the `workflow_dispatch` as there is no special parameter or execution flow and the previous builds can be always retriggered.

Reply:
I applied my suggestions but am happy to roll them back if you see problems with them.